hawk/app/models/cib.rb

```diff
@@ -1036,12 +1033,17 @@ def initialize(id, user, use_file = false, stonithwarning = false)
     # Now we can sort the node array
     @nodes.sort!{|a,b| a[:uname].natcmp(b[:uname], true)}
 
+    # Check CRM feature set of all nodes for consistency
     feature_sets = {}
     @nodes.each do |n|
-      if n.key? :crm_feature_set
-        fs = n[:crm_feature_set]
-        feature_sets[fs] ||= []
-        feature_sets[fs] << n[:name]
+      next unless n[:state] == :online && !(n[:remote] || n[:host])
+      if @xml.elements.each("cib/status/node_state[@uname='#{n[:uname]}']/transient_attributes/instance_attributes/nvpair[@name='#feature-set']") do |fs_node|
+        fs = fs_node.attributes['value']
+        (feature_sets[fs] ||= []) << n[:name]
+      end.empty?
+        # The feature set attribute is present since 3.15.1. If it is missing
+        # then the node must be running an earlier version.
+        (feature_sets['<3.15.1'] ||= []) << n[:name]
```

Comment:
Key `:crm_feature_set` is removed from the node information because it was determined incorrectly by the code. This means that the condition `n.key? :crm_feature_set` would always evaluate to false.